sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java

```diff
@@ -724,21 +796,43 @@ public EventProcessorClientBuilder initialPartitionEventPosition(
         return this;
     }
 
+    /**
+     * Sets the default starting position for each partition if a checkpoint for that partition does not exist in the
+     * {@link CheckpointStore}.
+     *
+     * <p>
+     * Only <strong>one overload</strong> of {@code initialPartitionEventPosition} should be used when constructing
+     * an {@link EventProcessorClient}.
+     * </p>
+     *
+     * @param initialEventPositionProvider Function that maps the given {@code partitionId} to an
+     *      {@link EventPosition}.
+     *
+     * @return The updated {@link EventProcessorClientBuilder} instance.
+     */
+    public EventProcessorClientBuilder initialPartitionEventPosition(
+        Function<String, EventPosition> initialEventPositionProvider) {
+        this.initialEventPositionProvider = initialEventPositionProvider;
```

Comment:
nit; do we need a null check?

Reply:
Oh yes. It totally slipped my mind again. Thanks for reminding me.